policies/api/http/logging.go

```diff
@@ -214,6 +214,48 @@ func (l loggingMiddleware) ValidateDataset(ctx context.Context, token string, d
 	return l.svc.ValidateDataset(ctx, token, d)
 }
 
+func (l loggingMiddleware) ViewDatasetByID(ctx context.Context, token string, datasetID string) (_ policies.Dataset, err error) {
+	defer func(begin time.Time) {
+		if err != nil {
+			l.logger.Warn("method call: view_dataset_by_id",
+				zap.Error(err),
+				zap.Duration("duration", time.Since(begin)))
+		} else {
+			l.logger.Info("method call: view_dataset_by_id",
+				zap.Duration("duration", time.Since(begin)))
+		}
+	}(time.Now())
+	return l.svc.ViewDatasetByID(ctx, token, datasetID)
+}
+
+func (l loggingMiddleware) ListDataset(ctx context.Context, token string, pm policies.PageMetadata) (_ policies.PageDataset, err error) {
+	defer func(begin time.Time) {
+		if err != nil {
+			l.logger.Warn("method call: list_dataset",
+				zap.Error(err),
+				zap.Duration("duration", time.Since(begin)))
+		} else {
+			l.logger.Info("method call: list_dataset",
+				zap.Duration("duration", time.Since(begin)))
+		}
+	}(time.Now())
+	return l.svc.ListDataset(ctx, token, pm)
+}
+
 func NewLoggingMiddleware(svc policies.Service, logger *zap.Logger) policies.Service {
 	return &loggingMiddleware{logger, svc}
 }
+
+func (l loggingMiddleware) ListDatasetByPolicyIDInternal(ctx context.Context, policyID string, ownerID string) (_ []policies.Dataset, err error) {
+	defer func(begin time.Time) {
+		if err != nil {
+			l.logger.Warn("method call: list_dataset_by_groups",
+				zap.Error(err),
+				zap.Duration("duration", time.Since(begin)))
+		} else {
+			l.logger.Info("method call: list_dataset_by_groups",
+				zap.Duration("duration", time.Since(begin)))
+		}
+	}(time.Now())
+	return l.svc.ListDatasetsByPolicyIDInternal(ctx, policyID, ownerID)
```

Comment:
this seems to be calling a different function, where  `Datasets` is plural with an s, seems like that maybe a typo, or at least an inconsistency?